### PR TITLE
Release notes for SmartRedis v0.3.1

### PR DESCRIPTION
**Pull request by billschereriii**
Release notes for SmartRedis version 0.3.1 release

### CODE REVIEW

**Review by mellis13 — APPROVED** [review]
Thanks for putting this together.  Just a couple of comments.  It's amazing to see how much work has gone into this release!

File: doc/changelog.rst [hunk 1]
```diff
@@ -1,10 +1,102 @@
...
+  - rename_list()
+  - get_list_length()
+  - poll_list_length()
+  - poll_list_length_gte()
+  - poll_list_length_lte()
+  - get_datasets_from_list()
+  - get_dataset_list_range()
+  - use_list_ensemble_prefix()
+
+ - Implement multithreaded execution of parallel tasks, including all multi-shard operations. The number of threads devoted for this purpose is controlled by the new environment variable SR_THERAD_COUNT. The value defaults to 4, but may be any positive integer or special value zero, which will cause the SmartRedis runtime to allocate one thread for each available hardware context. (PR251_) (PR246_)
```
Comment (ashao) [inline doc/changelog.rst]: If we go for a shorter, pithier style of release notes, I'd say reword this to:
"Many of the new operations related to dataset aggregation and multigpu functions can now be parallelized based on a user-specified number of threads"

File: doc/changelog.rst [hunk 2]
```diff
@@ -1,10 +1,102 @@
...
+  - set_model_from_file_multigpu()
+  - set_model_multigpu()
+  - set_script_from_file_multigpu()
+  - set_script_multigpu()
+  - run_model_multigpu()
+  - run_script_multigpu()
+  - delete_model_multigpu()
+  - delete_script_multigpu()
+
+ - Added API calls for all clients to delete models and scripts from the backend database. (PR240_) New commands are:
```
Comment (ashao) [inline doc/changelog.rst:38]: Probably eliminate the description of the new methods

File: doc/changelog.rst [hunk 3]
```diff
@@ -1,10 +1,102 @@
...
+  - delete_script()
+  - delete_model()
+
+ - Added clustered Redis testing to automated GitHub check-in testing. (PR239_)
```
Comment (ashao) [inline doc/changelog.rst]: Maybe the right way to organize this is by theme by either the portion of the code (e.g. 'Fortran Client') or by type of change (e.g. bug fix, performance enhancements, quality of life).

**Review by ashao — CHANGES_REQUESTED** [review]
I would recommend that we try to have a shorter set of release notes, though this level of detail might be right for the changelog itself. For release notes, highlighting the primary new functionality (i.e. the new threaded capabilities and the dataset aggregation) would make it more likely that we would have casual (or advanced) users understand the major improvements.

**Review by mellis13 — APPROVED** [review]
